public/firebase.js

```diff
@@ -0,0 +1,218 @@
+      window.open('#/','_self')
+    }
+  });
+}
+//Login facebook
+function loginFB(){
+  firebase.auth()
+  .signInWithPopup(providerFB)
+  .then(function(result) {
+    console.log(result.user);
+    saveDataUser(result.user);
+      window.open('#/','_self')
+  });
+}
+
+function createEmailPass(email, password, names) {
+  firebase.auth()
+    .createUserWithEmailAndPassword(email, password)
+    .then((result) => {
+      saveDataUser(result.user);
+      result.user.updateProfile({
+        displayName: names,
+      });
+      if (result.user.updateProfile){
+        window.open('#/','_self')
+      }
+    })
+    .catch((error) => {
+      console.error(error);
+      alert("ERROR");
+    });
+}
+
+function registerUser() {
+  const email = document.getElementById("email").value;
+  const password = document.getElementById("password").value;
+  const names = document.getElementById("names").value;
+
+  createEmailPass(email, password, names);
+}
+
+//Save user by login
+function saveDataUser(user){
+  let userNew = {
+    uid:user.uid,
+    name:user.displayName,
+    photo:user.photoURL
+  }
+  db.collection("users").doc(userNew.uid).set(userNew)
+  .then(function() {
+    console.log("Document successfully written!");
+  })
+  .catch(function(error) {
+    console.error("error save user", error);
+     });
+  }
+
+  //User login e-mail
+
+  function loginUser(email, password){
+    firebase.auth().signInWithEmailAndPassword(email, password)
+    .then(function(user){
+      window.open('#/', '_self');
+    }).catch(function(error) {
+      let errorCode = error.code;
+      let errorMessage = error.message;
+    });
+  }
+
+  function loginUserEmail(){
+    const email = document.getElementById("e-mail").value;
+    const password = document.getElementById("password").value;
+  
+    loginUser(email, password);
+  }
+
+  function signOff(){
+    firebase.auth().signOut().then(function() {
+      window.open('#/login', '_self');
+    }).catch(function(error) {
+    });
+    
+  } 
+
+//add post user
+function addUserPost(){
+  let user = firebase.auth().currentUser;
+  let post = document.getElementById("addNewPost").value;
+  db.collection("post").add({
+    post: post,
+    uid: user.uid,
+    user: user.displayName,
+    imagen: user.photoURL
+})
+.then(function(docRef) {
+  console.log("Document written with ID: ", docRef.id);  
+  document.getElementById("addNewPost").value = '';
+    showPostUser();
+    
+})
+.catch(function(error) {
+    console.error("Error adding document: ", error);
+});
+}
+
+//Show post user
+function showPostUser(){
+  const postContainer = document.getElementById('allNewPost');
+        postContainer.innerHTML='';
+        db.collection("post").onSnapshot((querySnapshot) => {
+        querySnapshot.forEach((doc) => {;
+        //console.log(`${doc.id} => ${doc.data().post}`);
+        let postElement = document.createElement('div');
+        let postNew = `
+        <div class="containerPostPrint">
```

Comment:
si la clase es para implementar estilos no se debe utilizar camelCase en vez de containerPostPrint, para estilos es mejor container-post-print
```suggestion
        <div class="container-post-print">
```